x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -60,6 +60,114 @@ func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx
 		SwapFee:                   sdk.NewDecWithPrec(1, 2),
 	}, nil
 }
+func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCreatePosition, error) {
+	// get random pool
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get random user address with the pool denoms
+	sender, tokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	// ensure that we have atleast 2 tokens
```

Comment:
```suggestion
	// ensure that we have at least 2 tokens
```
Would maybe mention something about this check being necessary since SelAddrWithDenoms selects a random subset of denoms. 

Sidenote, is there not a function that selects and address with denoms and returns the ENTIRE denom subset? If so, I feel like thats what we would want to use here. If not, it shouldn't be hard to create.

Reply:
I tried few stuff, and i think the closest efficient thing i go to was do is; 
```
userCoins := sim.BankKeeper().SpendableCoins(ctx, acc.Address)
checkSubset := poolDenoms.DenomsSubsetOf(userCoins)
if !checkSubset {
 error ("user doesnot contain pool denoms")
}
```

So the issue with this is that; 
1. poolDenoms is a denomString and doesnot contains full coinAmt as cannot retrieve that from CLPool
2. I created my own denom subset function that only checks if the account contains denom, but this requires multiple for loop and i'm afraid that it might slow down the simulator for such a simple task